### PR TITLE
[addons] fix PVR disable/enable handle

### PR DESCRIPTION
**Pull request by AlwinEsch**
## Description

Fix PVR disable/enable handle. Before it used the Binary addon manager about but with the PVR was informed about change before them and not worked.

Also remove the no more used IsAddonEnabled from CBinaryAddonManager.

## Types of change

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [ ] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [ ] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

**Review by ksooo — COMMENTED** [review]
Not sure, but looking at the changes of this PR and at what I saw while I was debugging issues after switching profiles containing PVR addons, for me it looks like binary addon manager's data getting out of sync somehow. 

So, instead of letting binary addon manager call addon manager instead of using its own data (because they are broken), seems more like a work around than a real fix.

I suggest to fix binary addon manager to contain consistent data again.

File: xbmc/guilib/imagefactory.cpp [hunk 1]
```diff
@@ -37,9 +36,9 @@ IImage* ImageFactory::CreateLoader(const CURL& url)
 
 IImage* ImageFactory::CreateLoaderFromMimeType(const std::string& strMimeType)
 {
-  BinaryAddonBaseList addonInfos;
+  std::vector<AddonInfoPtr> addonInfos;
```
Comment (ksooo) [inline xbmc/guilib/imagefactory.cpp:40]: Remove this empty line? At other code sequences similar to this one there is no blank line.

File: xbmc/peripherals/bus/virtual/PeripheralBusAddon.cpp [hunk 2]
```diff
@@ -410,9 +408,10 @@ void CPeripheralBusAddon::UpdateAddons(void)
   {
     CLog::Log(LOGDEBUG, "Add-on bus: Registering add-on %s", addonId.c_str());
 
-    auto GetAddon = [&addonId](const BinaryAddonBasePtr& addon) { return addon->ID() == addonId; };
+    auto GetAddon = [&addonId](const AddonInfoPtr& addon) { return addon->ID() == addonId; };
 
-    BinaryAddonBaseList::iterator it = std::find_if(newAddons.begin(), newAddons.end(), GetAddon);
+    std::vector<AddonInfoPtr>::iterator it =
```
Comment (ksooo) [inline xbmc/peripherals/bus/virtual/PeripheralBusAddon.cpp:413]: `auto it`?

**Review by ksooo — COMMENTED** [review]
Minors only. 

However, I cannot judge on the functional changes and there consequences. For example, removal of PVR addons from binary addon cache. But I trust you @AlwinEsch .

**Review by AlwinEsch — COMMENTED** [review]
Thanks for the review, have updated this parts.
Also changed the function name to uppercase in last commit.
Further is a new patch added where fix the Kodi start if addon auto update is disabled.

**Review by ksooo — APPROVED** [review]
LGTM now.

**Pull request merged**